src/dune_rules/lib.ml

```diff
@@ -950,16 +970,14 @@ end = struct
           let+ impl = impl in
           impl :: requires)
     in
-    let* ppx_runtime_deps =
-      Lib_info.ppx_runtime_deps info |> resolve_simple_deps db ~private_deps
-    in
     let src_dir = Lib_info.src_dir info in
-    let map_error x =
-      Resolve.push_stack_frame x ~human_readable_description:(fun () ->
-          Dep_path.Entry.Lib.pp { name; path = src_dir })
+    let ppx_runtime_deps_host =
+      Memo.lazy_ (fun () ->
+          Lib_info.ppx_runtime_deps info
+          |> resolve_simple_deps db ~private_deps
+          >>| add_stack_frame name ~path:src_dir)
```

Comment:
Can't you use the memoized function you've introduced to get the ppx runtime deps for the host?

Reply:
It's also a little weird that we first resolve the deps and then add the stack frame. Shouldn't it be the other way?

Reply:
> Can't you use the memoized function you've introduced to get the ppx runtime deps for the host?

OH I can probably do that, indeed. Let me tyr